server/src/main/java/org/elasticsearch/action/admin/indices/rollover/TransportRolloverAction.java

```diff
@@ -242,9 +248,18 @@ static String generateRolloverIndexName(String sourceIndexName, IndexNameExpress
             .collect(Collectors.toMap(result -> result.condition.toString(), result -> result.matched));
     }
 
-    static Map<String, Boolean> evaluateConditions(final Collection<Condition<?>> conditions, final IndexMetaData metaData,
-                                                    final IndicesStatsResponse statsResponse) {
-        return evaluateConditions(conditions, statsResponse.getIndex(metaData.getIndex().getName()).getPrimaries().getDocs(), metaData);
+    static Map<String, Boolean> evaluateConditions(final Collection<Condition<?>> conditions,
+                                                   @Nullable final IndexMetaData metaData,
+                                                   @Nullable final IndicesStatsResponse statsResponse) {
+        if (metaData == null) {
+            return conditions.stream().collect(Collectors.toMap(Condition::toString, cond -> false));
+        } else {
+            final DocsStats docsStats = Optional.ofNullable(statsResponse)
+                .map(stats -> stats.getIndex(metaData.getIndex().getName()))
```

Comment:
Did we throw NPE because statsResponse was null or it did not have docStats for the given index? Should we also need to protect the latter case here?

Reply:
We can't tell from the original stacktrace since everything was on the same line, but I took the opportunity to protect against either here.

Assuming the index stats are not null, `getPrimaries()` won't return null due to: https://github.com/elastic/elasticsearch/blob/b638cc101f16918c5e6d3bf1eb857b008be54a05/server/src/main/java/org/elasticsearch/action/admin/indices/stats/IndexStats.java#L98-L110

Even if there are no docStats for the index, where either statsResponse, indexStats, or docStats is null, this won't produce an NPE any more

Reply:
I think we should filter null as this transformation `indexStats -> indexStats.getPrimaries().getDocs()` will throw exception if indexStats is null?

Reply:
> I think we should filter null values as this transformation indexStats -> indexStats.getPrimaries().getDocs() will throw exception if indexStats is null?

`indexStats` can't be null because this is wrapped in the `Optional` monad and will short circuit to the `orElse` if any of the `map` calls return `null`

Reply:
Sorry, Lee. You're right. I misread these lines.